src/common/layers/otel.rs

```diff
@@ -0,0 +1,787 @@
+use opentelemetry::{
```

Comment:
as far as i can tell, this module appears to be...a vendored version of `tracing-opentelemetry`'s `OpenTelemetryLayer` type? why does this need to be part of `hyper`?

again, the purpose of `tracing` providing a facade layer is that the application can choose what `Layer`s to use to record the _instrumentation_ emitted by libraries like `hyper`. for example, if an application developer wanted to record traces, including `hyper`'s instrumentation, using opentelemetry, they would simply depend on the `tracing-opentelemetry` crate. this shouldn't have to be part of `hyper` itself in order to use the instrumentation in `hyper`.